engine/gctscript.go

```diff
@@ -0,0 +1,77 @@
+package engine
+
+import (
+	"fmt"
+	"path/filepath"
+	"sync/atomic"
+
+	"github.com/thrasher-corp/gocryptotrader/gctscript/vm"
+	log "github.com/thrasher-corp/gocryptotrader/logger"
+)
+
+const gctscriptManagerName = "gctscript"
+
+type gctScriptManager struct {
+	running  atomic.Value
```

Comment:
Prefer this better than using int32, need to make a to-do item to switch over our managers all to use this plus the new SubSystem const messages

Reply:
There is a slight decrease in performance when using it as atomic.Value is an interface but its about a 1-3ns/op average using it

Reply:
For the managers context, a 1-3ns/op decrease in performance is fine. We only use it to prevent races when starting/stopping or checking to see if a manager is running so it shouldn't be used frequently